google/Sender.js

```diff
@@ -23,35 +26,40 @@ function Sender(senderID, serverKey, type) {
     this.queued = [];
     this.acks = [];
 
-	// Set default port to Production
-	var fcmPort = Constants.FCM_SEND_PRODUCTION_PORT;
+    // Set default port to Production
+    let fcmPort = Constants.FCM_SEND_PRODUCTION_PORT;
 
-    if (type && type === Constants.FCM_DEVELOPMENT_IDX ) {
-		fcmPort = Constants.FCM_SEND_DEVELOPMENT_PORT;
+    if (type && type === Constants.FCM_DEVELOPMENT_IDX) {
+        fcmPort = Constants.FCM_SEND_DEVELOPMENT_PORT;
         this.connectionType = Constants.FCM_DEVELOPMENT_IDX;
-	}
+    }
 
-    var self = this;
+    const self = this;
 
-    this.client = new xmpp.Client({
-        jid: this.senderId + '@gcm.googleapis.com',
+    this.client = client({
+        username: this.senderId + '@fcm.googleapis.com',
         password: this.serverKey,
-        port: fcmPort,
-        host: Constants.FCM_SEND_ENDPOINT,
-        reconnect: true,
-        legacySSL: true,
-        preferredSaslMechanism: Constants.FCM_PREFERRED_SASL
+        service: `xmpps://${Constants.FCM_SEND_ENDPOINT}:${fcmPort}`,
+        domain: 'fcm.googleapis.com'
+        // reconnect: true,
     });
+    // this.client.middleware.use((ctx, next) => {
+    //     console.log('use: ' + ctx.stanza.toString())
+    //    next();
+    // });
+    // this.client.middleware.filter((ctx, next) => {
+    //     console.log('filter: ' + ctx.stanza.toString())
+    // });
+    if (debug) {
+        xmppDebug(this.client, true);
+    }
 
-    this.client.connection.socket.setTimeout(0);
-    this.client.connection.socket.setKeepAlive(true, 10000);
-
-    this.client.on('online', function () {
+    this.client.on('online', async (address) => {
         self.events.emit('connected', self.connectionType);
 
         if (self.draining) {
             self.draining = false;
-            var i = self.queued.length;
+            let i = self.queued.length;
             while (i--) {

```

Comment:
are you sure `let i` is mutable here?

Reply:
Yes ofcourse it is. 
https://stackoverflow.com/questions/762011/whats-the-difference-between-using-let-and-var